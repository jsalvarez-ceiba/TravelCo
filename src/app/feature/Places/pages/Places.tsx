import React, { useEffect, useCallback } from 'react';
import { useDispatch } from 'react-redux';
import * as PropTypes from 'prop-types';

import ContainerPlaces from '../components/ContainerPlaces/ContainerPlaces';
import { getPlaces } from 'app/core/redux/actions/places/placesActions';
import Footer from '../../Footer/Footer';

export const Places: React.FC = () => {
  const dispatch = useDispatch();
  const getList = useCallback(() => {
    dispatch(getPlaces());
  }, [dispatch]);

  useEffect(() => {
    getList();
  }, [getList]);

  return (
    <>
    <div className="flexCenter">
      <ContainerPlaces />
    </div>
    <Footer />
    </>
  );
};

Places.propTypes = {
  getPlaces: PropTypes.func.isRequired,
};
