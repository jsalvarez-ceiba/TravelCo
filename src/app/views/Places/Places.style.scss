.background {
    background-image: url('../../../assets/img/travel.jpg');
    background-size: cover;
    background-repeat: no-repeat;
    width: 100vw;
    height: 100vh;
    overflow-x: hidden;
}
.flexCenter {
    display: flex;
    justify-content: center;
    align-items: center;
    height: auto;
}
.title {
    color: white;
}