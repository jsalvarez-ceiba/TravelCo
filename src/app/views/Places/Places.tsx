import React, { useEffect } from 'react';
import ContainerPlaces from '../../components/ContainerPlaces/ContainerPlaces';
import './Places.style.scss';
//import { useSelector } from 'react-redux';
//import { NotesState } from 'app/core/redux/reducers/reservationReducer';



const Places = () => {
    /* const notes = useSelector<NotesState, NotesState['notes']>((state) => state.notes); */

    useEffect(() => {
        /* console.log('notes : ', notes); */
    }, []);

    return (
        <div className="background">
            <div data-aos="fade-down" data-aos-easing="linear" data-aos-duration="1500" className="flexCenter p-3">
                <h1 className="title">Destinos Nacionales e Internacionales</h1>
            </div>
            <div className="flexCenter mt-5">
                <ContainerPlaces />
            </div>
        </div>
    );
};

export default Places;
