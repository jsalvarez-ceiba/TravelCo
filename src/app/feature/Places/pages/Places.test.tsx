import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import { Places } from './Places';
import configureMockStore from 'redux-mock-store';
import { render } from '@testing-library/react';
import thunk from 'redux-thunk';

const middlewares = [thunk];

const mockStore = configureMockStore(middlewares);

describe('Test Places View', () => {
  const initialState = {};
  let store, wrapper;

  test('render component', () => {
    store = mockStore(initialState);

    const component = mount(
      <Provider store={store}>
        <Places />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });

  test('render div background', () => {
    store = mockStore(initialState);
    const component = mount(

      <Provider store={store}>
        <Places />
      </Provider>

    );
    expect(component.find('h1')).toEqual({});

  })
});
