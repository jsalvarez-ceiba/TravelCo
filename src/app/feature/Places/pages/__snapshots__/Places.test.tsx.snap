// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Places View render component 1`] = `
Array [
  <div
    className="flexCenter"
  >
    <div
      className="flexCenter background"
    >
      <h1>
        cargando...
      </h1>
    </div>
  </div>,
  <div
    className="flexCenter bg-dark"
  >
    <small
      className="text-white"
    >
      TravelCo | Todos los derechos reservados 2021
    </small>
  </div>,
]
`;
